### ISSUE TITLE
Global Container Security Context does not work for Git Sync container

### ISSUE THREAD

**Issue by ahipp13** [apache/airflow#35350]
### Official Helm Chart version

1.10.0

### Apache Airflow version

2.6.3

### Kubernetes Version

1.26.5

### What happened

Idk if this is the intended way this should work, but when I set the global containerSecurityContext like:
```
# Detailed default security context for airflow deployments
securityContexts:
  pod:
    runAsUser: 50000
    runAsGroup: 50000
    fsGroup: 50000
  containers:
    capabilities:
      drop:
        - KILL
        - MKNOD
        - SYS_CHROOT
 ```
        
 This does not apply to the "git sync" container. In order for me to get it to apply to the git sync container, I also have to set these in the "dags.gitSync" section like:
 ```
     securityContexts:
      container:
        capabilities:
          drop:
            - KILL
            - MKNOD
            - SYS_CHROOT
```
### What you think should happen instead

When you set a global container security context, it should also apply to the git sync container

### How to reproduce

The cluster I deploy to requires some container security context settings, so for me to reproduce I just set the global container security context and try to deploy, and if it won't let me deploy I look in the logs and it will say the git sync container does not have the required container security context.

### Anything else

I looked through the 1.11.0 helm chart release notes and did not see this was fixed. Also I do not know if this is the intended way this is supposed to work. Just wanted to bring it to light!

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by JKrehling** [comment]
@shahar1  Do you know if there is currently an ongoing plan to make airflow compatible with readOnlyRootFilesystem?
And would there be any objection to adding the restrictive securityContexts by default?

**Comment by potiuk** [comment]
> @shahar1 Do you know if there is currently an ongoing plan to make airflow compatible with readOnlyRootFilesystem?
And would there be any objection to adding the restrictive securityContexts by default?

Just to add here - I don't think there are "plans" or roadmap (other than big Airflow 3 Airflow Improvement Proposals we have) - for things like this, if anyone wants to get this done, they are most welcome to implement it, second-best is finding somoene and making them implement it. Other than that, there is no-one who "knows" if other people have plans like that - generally if somone submits a change to make airflow works with `readOnlyRootFilesystem` and it will pass the review, then it will be done. But we (maintainers) do not know if any of the almost 3000 contributors will submit it.

So the best way to speed it up if someone needs it is to .... contribute it.

**Comment by potiuk** [comment]
I marked it as "good first issue" and hopefully that might attract somoene who will implement it.

**Comment by JKrehling** [comment]
Fair enough.  I only ask because last time I went to submit a similar MR for another project there was already work being done to implement it.  

If its not done by the time I get this working on my system I'll try and put together an MR

**Comment by potiuk** [comment]
> Fair enough. I only ask because last time I went to submit a similar MR for another project there was already work being done to implement it.

Generally speaking any of us can do search and find out (in PRs or issues) - because absolutely no-one can keep it all in a head - this is why we have GitHub discussions here. And there is notthing with asking (opening a discussion for example) in general - but tasking a specific maintainer to give answer is likely not helping in getting an answer - because by doing it you limit your chances to get answer from somone who knows - because they will think the tagged person will (but that person might or might not know anything about and you implicitly task (and delegate it) to that person in looking for an answer for you if they do not know. That's not too nice for those people who often contribute here as volunteers. No bad feelings - just want to explain how it works here.

In other words - I am just saying - if you want to check if something is being done, best thing is to do search - here and in our wiki (see community link) and second best is ask a generic question without tagging anyone specific and hope somoene will be attracted and answer you.

> You can see the logic controlling this around line 316 of the operator in the PR.

Cool !
